privateSBOMExchange/src/petra/lib/models/tree_ops.py

```diff
@@ -172,11 +184,15 @@ def get_membership_proof(root, target_hash):
     def get_prefix(node):
         """This is same prefix used by MerkleVisitor during hashing"""
         if isinstance(node, SbomNode):
-            prefix = (f"{node.purl}").encode() 
+            prefix = node.redacted_keys+(f"{node.purl}").encode("utf-8") +node.plaintext_hash 
         elif isinstance(node, ComplexNode):
-            prefix = (f"{node.encrypted_data}{node.complex_type}{node.policy}").encode()
+            prefix = (f"{node.encrypted_data}{node.policy}{node.complex_type}").encode("utf-8")+node.plaintext_commit.value+node.plaintext_hash
         else: #field node
-            prefix = b''
+            prefix = node.encrypted_data.encode("utf-8")+node.policy.encode("utf-8")
+            if node.encrypted_data != NODE_PUBLIC:
+                prefix += NODE_REDACTED.encode("utf-8")
+            else:
+                prefix += (f"{node.field_name}:{node.field_value}").encode("utf-8")
```

Comment:
Same here. FieldNodes now support serialize_for_hashing() so we ensure that we're always hashing the node contents in a consistent order and format. Please call that function here instead.

Reply:
I can use this one, but the complex serialization won't work because it appends the children